src/main/java/com/notably/logic/parser/suggestion/NewSuggestionCommandParser.java

```diff
@@ -38,12 +43,22 @@ public NewSuggestionCommandParser(Model model) {
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(userInput, PREFIX_TITLE, PREFIX_BODY, PREFIX_JUMP);
 
-        String title;
+        if (userInput.isBlank()) {
+            model.setResponseText(RESPONSE_MESSAGE);
+            return Optional.empty();
+        }
+
         if (!ParserUtil.arePrefixesPresent(argMultimap, PREFIX_TITLE)
                 || !argMultimap.getPreamble().isEmpty()) {
-            title = userInput.trim();
-        } else {
-            title = argMultimap.getValue(PREFIX_TITLE).get();
+            model.setResponseText(String.format(ERROR_MESSAGE, model.getInput()));
+            return Optional.empty();
+        }
+
+        String title = argMultimap.getValue(PREFIX_TITLE).get();
+
+        if (!Title.isValidTitle(title) && !title.isBlank()) {
+            model.setResponseText(String.format(ERROR_MESSAGE_INVALID_TITLE, title));
+            return Optional.empty();
         }
```

Comment:
Hmm why not combine this with the `if else` group below?

Reply:
Ok!